docs/proposals/2017-04-20 - Charts.md

```diff
@@ -0,0 +1,201 @@
+# Problem
+
+Talking to publishing support, it seems as though the most often use case for tables was when doing a "basic data chart."  Estimated usage is 1-2 times a week throughout the entire newsroom. The following are some recent examples:
+
+https://www.washingtonpost.com/lifestyle/kidspost/who-deserves-a-shot-at-the-nba-all-star-game/2017/01/25/c9995306-de5e-11e6-918c-99ede3c8cafa_story.html
+
+https://www.washingtonpost.com/news/capital-weather-gang/wp/2017/03/02/does-a-record-warm-winter-mean-a-super-hot-summer-in-d-c/
+
+Ellipsis needs to create data charts like these within an article. They need to add borders on rows or columns, a citation, and align data in cells.  Rows and columns may or may not be titled.
+
+The newsroom also uses Methode/HTML tables for other purposes via templates -- notably timelines and bar charts. These are used less often and are each tied to specific template. For Ellipsis and ANS purposes, they should be considered a separate use case. See [ANS Stat Series](2017-04-20 - Statistic Series.md).
+
+The following proposal deals specifically with the use case of creating a basic data chart.
+
+# Proposal
+
+A new `chart` format that includes row/column settings and a citation, and models cells as text elements (rather than strings.)
+
+## Title
+
+An ANS `text` element.
+
+## Citation
+
+An ANS `text` element, identical to a `citation` on the new `quote` element.  Intended to indicate the source of the data.
+
+## Credits
+
+An ANS credits object, can denormalize authors from author service. Intended to indicate the person who created this specific chart.
+
+## No width/height/sizing options
+
+Explicit sizing options are deliberately not included.
+
+## Explicitly Defined Columns - `columns`
+
+Data columns should be explicitly defined in the `columns` attribute. The following attributes can be set:
+
+* `_id` Automatically-generated.
+* `text_alignment` can be optionally set and applies to the entire column.
+* `title` text element can be optionally supplied here which can function as a header for a column.  **If at least one column has a title than a header row containing the title should be added to the chart. Otherwise it should be omitted.**
+* `borders` an array of strings with the possible values `"left"` and `"right"`
+
+## Explicitly Defined Rows - `rows`
+
+* `_id` Automatically-generated.
+* `title` text element can be optionally supplied here which can function as a header for a row.  **If at least one row has a title then a header column containing the title should be added to the chart. Otherwise it should be omitted.**
+* `borders` an array of strings with the possible values `"top"` and `"bottom"`
+* `cells` an array of cell objects.
+
+## Cells
+
+Currently these objects have just two properties:
+
+* `_id` - Automatically-generated
+* `cell_content` - currently this object can only be an ANS `text` element.
```

Comment:
I can see the argument for having content be an array of `content_elements`.  It feels kind of unfortunate, just because it adds some bloat and will be unused in most cases.

Do we absolutely need `raw_html` here?  I think we'd like to relegate that option to only when and where absolutely necessary, otherwise we will end up with it everywhere.  And it seems like the content of a chart should not be so complex that it can't be represented by text elements.